source/databases-collections.txt

```diff
@@ -0,0 +1,286 @@
+by passing an options array to the ``MongoDB\Client::selectCollection()`` method.
+
+To change the read preference, read concern, and write concern of your database
+or collection, set the following options in the array parameter:
+
+- ``readPreference``: Sets the read preference. For a list of available read
+  preferences, see `MongoDB\\Driver\\ReadPreference <{+php-manual+}/class.mongodb-driver-readpreference>`__
+  in the extension API documentation.
+- ``readConcern``: Sets the read concern. For a list of available read
+  concerns, see `MongoDB\\Driver\\ReadConcern <{+php-manual+}/class.mongodb-driver-readconcern>`__
+  in the extension API documentation.
+- ``writeConcern``: Sets the write concern. For a list of available write
+  concerns, see `MongoDB\\Driver\\WriteConcern <{+php-manual+}/class.mongodb-driver-writeconcern>`__
+  in the extension API documentation.
+
+.. tip::
+
+   To learn more about read preferences and read and write concerns, see the
+   following guides in the MongoDB Server manual:
+
+   - :manual:`Read Preference </core/read-preference/>`
+   - :manual:`Read Concern </reference/read-concern/>`
+   - :manual:`Write Concern </reference/write-concern/>`
+
+Database Configuration Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example shows how to set the read preference, read concern, and
+write concern of a database called ``test_database`` by passing an options
+array to ``selectDatabase()``:
+
+.. literalinclude:: /includes/databases-collections/databases-collections.php
+    :language: php
+    :dedent:
+    :start-after: start-database-settings
+    :end-before: end-database-settings
+
+Collection Configuration Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example shows how to set the read preference, read concern, and
+write concern of a collection called ``test_collection`` by passing an options
+array to ``selectCollection()``:
+
+.. literalinclude:: /includes/databases-collections/databases-collections.php
+    :language: php
+    :dedent:
+    :start-after: start-collection-settings
+    :end-before: end-collection-settings
+
+Tag Sets
+~~~~~~~~
+
+In {+mdb-server+}, you can apply key-value :manual:`tags
+</core/read-preference-tags/>` to replica-set
+members according to any criteria you choose. You can then use
+those tags to target one or more members for a read operation.
+
+By default, the {+php-library+} ignores tags when choosing a member
+to read from. To instruct the {+php-library+} to prefer certain tags,
+pass them as a parameter to your ``MongoDB\Driver\ReadPreference`` class
+constructor. Then, set the ``MongoDB\Driver\ReadPreference`` object as
+the value of the ``readPreference`` database option.
+
+The following code example sets the ``readPreference`` option to a tag set
+that instructs ``test_database`` to prefer reads from the New York data
+center (``'dc' => 'ny'``) and to fall back to the San Francisco data
+center (``'dc' => 'sf'``):
```

Comment:
It's been a while since I revisited tag set docs, but I recall examples typically included an empty tag set as a fallback.

As-is, server selection will fail if there are no servers in either `dc:ny` or `dc:sf`. Just want to confirm that is intentional.

Reply:
Added an empty tag set